cmd/acra-translator/server/server.go

```diff
@@ -193,129 +187,216 @@ func (server *ReaderServer) HandleConnectionString(parentContext context.Context
-			if err != nil {
-				log.WithError(err).WithField(logging.FieldKeyEventCode, logging.EventCodeErrorTranslatorCantHandleHTTPConnection).
-					Errorln("Took error on handling HTTP requests")
-				server.Stop()
-				os.Exit(1)
-			}
-		}()
+		// create HTTP listener from correspondent file descriptor
+		file := os.NewFile(fdHTTP, "/tmp/acra-translator_http")
+		if file == nil {
+			logger.Errorln("Can't create new file from descriptor for acra listener")
+			return
+		}
+		listenerFile, err := net.FileListener(file)
+		if err != nil {
+			logger.WithError(err).WithField(logging.FieldKeyEventCode, logging.EventCodeErrorCantOpenFileByDescriptor).
+				Errorln("System error: can't start listen for file descriptor")
+			return
+		}
+
+		listenerWithFileDescriptor, ok := listenerFile.(network.ListenerWithFileDescriptor)
+		if !ok {
+			logger.WithError(err).WithField(logging.FieldKeyEventCode, logging.EventCodeErrorFileDescriptionIsNotValid).
+				Errorf("System error: file descriptor %d is not a valid socket", fdHTTP)
+			return
+		}
+		server.listenerHTTP = listenerWithFileDescriptor
+		server.startHTTP(parentContext, logger, decryptorData, errCh, listenerWithFileDescriptor)
 	}
+
 	// provide way to register new services and custom server
 	if server.config.IncomingConnectionGRPCString() != "" {
-		server.waitGroup.Add(1)
-		go func() {
-			defer server.waitGroup.Done()
-			grpcLogger := logger.WithField(ConnectionTypeKey, GRPCConnectionType)
-			logger.WithField("connection_string", server.config.IncomingConnectionGRPCString()).Infof("Start process gRPC requests")
-			var listener net.Listener
-			var err error
-			var opts []grpc.ServerOption
-			if server.config.WithTLS() {
-				listener, err = network.Listen(server.config.IncomingConnectionGRPCString())
-				if err != nil {
-					grpcLogger.WithError(err).WithField(logging.FieldKeyEventCode, logging.EventCodeErrorTranslatorCantHandleGRPCConnection).
-						Errorln("Can't create gRPC connection listener")
-					return
-				}
-				opts = append(opts, grpc.Creds(credentials.NewTLS(server.config.GetTLSConfig())))
-			} else {
-				listener, err = network.Listen(server.config.IncomingConnectionGRPCString())
-				if err != nil {
-					grpcLogger.WithError(err).Errorln("Can't initialize connection listener")
-					return
-				}
-				wrapper, err := network.NewSecureSessionConnectionWrapper(server.config.ServerID(), server.keystorage)
-				if err != nil {
-					grpcLogger.WithError(err).Errorln("Can't initialize Secure Session wrapper")
-					return
-				}
-				opts = append(opts, grpc.Creds(wrapper))
-			}
-
-			grpcListener := common.WrapListenerWithMetrics(listener)
-
-			grpcServer, err := server.grpcServerFactory.New(decryptorData, opts...)
+		// load gRPC listener from correspondent file descriptor
+		file := os.NewFile(fdGRPC, "/tmp/acra-translator_grpc")
```

Comment:
While we're here, maybe the `"/tmp/acra-translator_grpc"` should be extracted into some constant. This value is actually irrelevant, it's only a placeholder. But would be nice to indicate to the reader that it's not some random value. `grpcFilenamePlaceholder`? Does anyone have better ideas?

Reply:
Yes. It would be better. Thank you